docs: update client concept guides for the 2026-07-28 protocol changes (#3067)
modelcontextprotocol/modelcontextprotocol: docs/docs/draft/tools/debugging.mdx

</Warning>

> For servers using the
> [Streamable HTTP transport](/specification/draft/basic/transports/streamable-http),
> stderr is not captured by the client. Use your own server-side log aggregation
> or [OpenTelemetry](https://opentelemetry.io/), or standard HTTP tooling (curl,
> browser DevTools Network panel) to inspect requests and SSE streams.
> 
> <Warning>
> 
> Logging is deprecated as of protocol version `2026-07-28`. It remains
> available during the deprecation window, but new servers should log to stderr
> (stdio transport) or use [OpenTelemetry](https://opentelemetry.io/) instead.
> 
> </Warning>

Reviewer: This Warning restates the stderr/OpenTelemetry advice from the paragraph right above it, so the same guidance lands twice back to back. Tighter version that just names what's deprecated:

```suggestion
For servers using the
[Streamable HTTP transport](/specification/draft/basic/transports/streamable-http),
stderr is not captured by the client. Use your own server-side log aggregation
or [OpenTelemetry](https://opentelemetry.io/) for logs, and standard HTTP
tooling (curl, browser DevTools Network panel) to inspect requests and SSE
streams.

<Warning>

The `notifications/message` mechanism below is deprecated as of protocol
version `2026-07-28`. It remains available during the deprecation window.

</Warning>
```
